### ISSUE TITLE
안녕하세요. 커스텀 데이터셋을 만들었는데요.

### ISSUE THREAD

**Issue by chartgod** [SEUNKOREA/Llama2-7b-ko-FT#1]
파인튜닝.py를 사용했는데 checkpoint 디렉터리에 저장이 되었습니다.
파일에는 adapter_model.bin , optimizer.pt , rng_state.pth , scheduler.pt , training_args.bin 이 있는데 
학습 완료된 모델 가중치 파일이 adapter_model.bin이 맞나요?

**Comment by seunkorea** [comment]
안녕하세요. 

`finetuning.py`는 4bit-QLoRA 방식으로 fine-tuning을 진행하는 것이기 때문에 adapter layer만 학습이 진행됩니다. 따라서 checkpoint 디렉토리에 저장되는 가중치 파일은 adapter layer의 가중치만 저장됩니다!

따라서 fine-tuning 이후에 원본 가중치와 합쳐주어야 나중에 체크포인트를 불러와서 사용할 수 있습니다! 원본 가중치와 합치는 부분은  `finetuning.py` 에 `### Merge weights` 부분부터 입니다!

혹시나 더 궁금한 점이 있으시면 편하게 질문해주세요 !

**Comment by chartgod** [comment]
감사합니다. 혹시 데이터셋 만들기 위해 크롤링을 따로 하셨나요?

**Comment by seunkorea** [comment]
안녕하세요.

데이터셋을 만들기 위해서 크롤링은 따로 진행하지 않았고
KorQuAD 2.0 텍스트 파일을 모두 합친 후에 HTML 태그 제거정도의 전처리 과정을 거친후에
허깅페이스 데이터 포맷으로 변경하여 사용하였습니다 :)
